connectors/file/translator-parquet/src/main/java/org/teiid/translator/parquet/BaseParquetExecution.java

```diff
@@ -0,0 +1,358 @@
+    private long pageRowCount;
+
+    public BaseParquetExecution(ExecutionContext executionContext,
+                                RuntimeMetadata metadata, VirtualFileConnection connection, boolean immutable) {
+        this.executionContext = executionContext;
+        this.metadata = metadata;
+        this.connection = connection;
+        this.immutable = immutable;
+    }
+
+    public void visit(LanguageObject command) throws TranslatorException {
+        this.visitor.visitNode(command);
+
+        if (!visitor.exceptions.isEmpty()) {
+            throw visitor.exceptions.get(0);
+        }
+    }
+
+    @Override
+    public void execute() throws TranslatorException {
+        String path = this.visitor.getParquetPath();
+        if(this.visitor.getTable().getProperty(ParquetMetadataProcessor.PARTITIONING_SCHEME) != null && this.visitor.getTable().getProperty(ParquetMetadataProcessor.PARTITIONING_SCHEME).equals("directory")){
+            path += getDirectoryPath(this.visitor.getColumnPredicates(), this.visitor.getTable().getProperty(ParquetMetadataProcessor.PARTITIONED_COLUMNS));
+        }
+        this.parquetFiles = VirtualFileConnection.Util.getFiles(path, this.connection, true);
+        VirtualFile nextParquetFile = getNextParquetFile();
+        if (nextParquetFile != null) {
+            this.rowIterator = readParquetFile(nextParquetFile);
+        }
+    }
+
+    private String getDirectoryPath(HashMap<String, Comparison> predicates, String partitionedColumns) throws TranslatorException {
+        String path = ""; // because we are only supporting the equality comparison as of now, otherwise we would return a list of paths to iterate over
+        String[] partitionedColumnsArray = getPartitionedColumns(partitionedColumns);
+        for(int i = 0; i < partitionedColumnsArray.length; i++){
+          if(predicates.containsKey(partitionedColumnsArray[i])) {
+              Comparison predicate = predicates.get(partitionedColumnsArray[i]);
+              predicates.remove(partitionedColumnsArray[i]);
+              Literal l = (Literal) predicate.getRightExpression();
+              path += "/" + partitionedColumnsArray[i] + "=";
+              switch (predicate.getOperator()) {
+                  case EQ:
+                      switch (l.getType().getName()){
+                          case "java.lang.String":
+                          case "java.math.BigInteger":
+                          case "java.lang.Long":
+                          case "java.lang.Integer":
+                          case "java.lang.Boolean":
+                              path += l.getValue().toString();
+                              break;
+                          default:
+                              throw new TranslatorException("Only string, biginteger, long, integer, boolean are supported.");
+                      }
+                      break;
+                  default: throw new TranslatorException("Only equal comparison is allowed");
```

Comment:
How about deleting only those elements in the map which have equality comparison and leave others for the filtering? This shall work if the partitioned columns are also in the file.

Reply:
> This shall work if the partitioned columns are also in the file.

Can we confirm if this is generally the case?

If so, then yes you can absolutely just let them be filters instead of part of the path.